lib/http/response.rb

```diff
@@ -92,13 +92,18 @@ def flush
     #   (not an integer, e.g. empty string or string with non-digits).
     # @return [Integer] otherwise
     def content_length
+      # http://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.3.3
+      # Clause 3: "If a message is received with both a Transfer-Encoding
+      # and a Content-Length header field, the Transfer-Encoding overrides the Content-Length.        
+      return nil if headers[Headers::TRANSFER_ENCODING]
+
       value = @headers[Headers::CONTENT_LENGTH]
       return unless value
 
       begin
         Integer(value)
       rescue ArgumentError
-        nil
+        raise(HeaderSyntaxError, 'wrong Content-Length format')
```

Comment:
@britishtea how would it prevent the response from being instantiated? `#content_length` is an instance method from it, therefore you need the instance to call the method, right?